NextcloudTalk/UserProfileTableViewController.swift

```diff
@@ -0,0 +1,941 @@
+            setupFieldsForScopeSelectionDialog(field: &field, currentValue: &currentValue, title: &title,
+                                               fieldValue: kUserProfilePhoneScope, currentValueText: account.phoneScope, titleValue: "Phone number")
+        } else if sender.tag == kAddressTextFieldTag {
+            setupFieldsForScopeSelectionDialog(field: &field, currentValue: &currentValue, title: &title,
+                                               fieldValue: kUserProfileAddressScope, currentValueText: account.addressScope, titleValue: "Address")
+        } else if sender.tag == kWebsiteTextFieldTag {
+            setupFieldsForScopeSelectionDialog(field: &field, currentValue: &currentValue, title: &title,
+                                               fieldValue: kUserProfileWebsiteScope, currentValueText: account.websiteScope, titleValue: "Website")
+        } else if sender.tag == kTwitterTextFieldTag {
+            setupFieldsForScopeSelectionDialog(field: &field, currentValue: &currentValue, title: &title,
+                                               fieldValue: kUserProfileTwitterScope, currentValueText: account.twitterScope, titleValue: "Twitter")
+        } else if sender.tag == kAvatarScopeButtonTag {
+            setupFieldsForScopeSelectionDialog(field: &field, currentValue: &currentValue, title: &title,
+                                               fieldValue: kUserProfileAvatarScope, currentValueText: account.avatarScope, titleValue: "Profile picture")
+        }
+        var options = [DetailedOption]()
+        let privateOption = setupDetailedOption(identifier: kUserProfileScopePrivate,
+                                                imageName: "mobile-phone", title: "Private",
+                                                subtitle: "Only visible to people matched via phone number integration", selected: currentValue == kUserProfileScopePrivate)
+        let localOption = setupDetailedOption(identifier: kUserProfileScopeLocal,
+                                              imageName: "password-settings", title: "Local",
+                                              subtitle: "Only visible to people on this instance and guests", selected: currentValue == kUserProfileScopeLocal)
+        let federatedOption = setupDetailedOption(identifier: kUserProfileScopeFederated,
+                                                  imageName: "group", title: "Federated",
+                                                  subtitle: "Only synchronize to trusted servers", selected: currentValue == kUserProfileScopeFederated)
+        let publishedOption = setupDetailedOption(identifier: kUserProfileScopePublished,
+                                                  imageName: "browser-settings", title: "Published",
+                                                  subtitle: "Synchronize to trusted servers and the global and public address book", selected: currentValue == kUserProfileScopePublished)
```

Comment:
In method setupDetailedOption I called NSLocalizedString and passed strings from parameters, here:
Because without that, in every line above we should call NSLozalizedString... and lines will be longer. Instead of that in function bellow, we pass string and in function body we use that string for NSLocalizedString.
    // Setup Dialog Options
    func setupDetailedOption(identifier: String, imageName: String, **title**: String, **subtitle**: String, selected: Bool) -> DetailedOption {
        let detailedOption = DetailedOption()
        detailedOption.identifier = identifier
        detailedOption.image = UIImage(named: imageName)?.withRenderingMode(.alwaysTemplate)
        detailedOption.title = NSLocalizedString(**title**, comment: "")
        detailedOption.subtitle = NSLocalizedString(**subtitle**, comment: "")
        detailedOption.selected = selected
        return detailedOption
    }

Reply:
Oh okay. I missed the part at `setupDetailedOption` sorry 😅
Since we don't have any comments for translators in those strings, I guess we can leave it like this for now.

Reply:
Deal